src/net/tcp.rs

```diff
@@ -7,12 +7,16 @@
 //!
 /// [portability guidelines]: ../struct.Poll.html#portability
 
+#[cfg(not(target_os = "redox"))]
```

Comment:
Could you explain on why this file needs changes?

I would prefer to avoid adding platform cfg guards in `src/net/tcp`.

Reply:
Back when I wrote this, I would argue that "redox doesn't support creating non-blocking sockets before an address is bound". I am extremely certain that I tried it, but... I've seen the source code of redox' libc now, and it seems to disagree.

It's still always better to use native rust where possible (redox' backend doesn't use libc), but I cannot safely say the old code won't wory. I'll try it again, later.

Reply:
Just double checked, also checked the code. There *are* null sockets implemented, but they don't support fcntl. I'll ask Jeremy what he'd like to do here.